libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -0,0 +1,1929 @@
+
+    # check if the index exists
+    if not _index_exists(connection, idx_name):
+        with connection.cursor() as cursor:
+            cursor.execute(ddl)
+        logger.info("Index created successfully...")
+    else:
+        logger.info("Index already exists...")
+
+
+@_handle_exceptions
+async def acreate_index(
+    client: Any,
+    vector_store: OracleVS,
+    params: Optional[dict[str, Any]] = None,
+) -> None:
+    async def context(connection: Any) -> None:
+        if params:
+            if params["idx_type"] == "HNSW":
+                await _acreate_hnsw_index(
+                    connection,
+                    vector_store.table_name,
+                    vector_store.distance_strategy,
+                    params,
+                )
+            elif params["idx_type"] == "IVF":
+                await _acreate_ivf_index(
+                    connection,
+                    vector_store.table_name,
+                    vector_store.distance_strategy,
+                    params,
+                )
+            else:
+                await _acreate_hnsw_index(
+                    connection,
+                    vector_store.table_name,
+                    vector_store.distance_strategy,
+                    params,
+                )
+
+        else:
+            await _acreate_hnsw_index(
+                connection,
+                vector_store.table_name,
+                vector_store.distance_strategy,
+                params,
+            )
+
+    await _handle_context(client, context)
+    return
+
+
+@_handle_exceptions
+async def _acreate_hnsw_index(
+    connection: AsyncConnection,
+    table_name: str,
+    distance_strategy: DistanceStrategy,
+    params: Optional[dict[str, Any]] = None,
+) -> None:
+    idx_name, ddl = _get_hnsw_index_ddl(table_name, distance_strategy, params)
+
+    # check if the index exists
+    if not await _aindex_exists(connection, idx_name):
+        with connection.cursor() as cursor:
+            await cursor.execute(ddl)
+            logger.info("Index created successfully...")
+    else:
+        logger.info("Index already exists...")
+
+
+@_handle_exceptions
+async def _acreate_ivf_index(
+    connection: AsyncConnection,
+    table_name: str,
+    distance_strategy: DistanceStrategy,
+    params: Optional[dict[str, Any]] = None,
+) -> None:
+    idx_name, ddl = _get_ivf_index_ddl(table_name, distance_strategy, params)
+
+    # check if the index exists
+    if not await _aindex_exists(connection, idx_name):
+        with connection.cursor() as cursor:
+            await cursor.execute(ddl)
+        logger.info("Index created successfully...")
+    else:
+        logger.info("Index already exists...")
+
+
+@_handle_exceptions
+def drop_table_purge(client: Any, table_name: str) -> None:
+    """Drop a table and purge it from the database.
+
+    Args:
+        client: The OracleDB connection object.
```

Comment:
Use 'oracledb' connection object I.e. with the correct, lowercase python-oracledb module name.